app/pages/learn/01_tutorial/04_mastering-the-api/02_invoke/00_methodhandle.md

```diff
@@ -0,0 +1,626 @@
+we provide the reference to the target array, the index of an element, and the new value:
+
+```java
+MethodHandle elementSetter = MethodHandles.arrayElementSetter(String[].class);
+elementSetter.invoke(arr, 4, "test");
+```
+
+To read the value of a single element, we should use [`arrayElementGetter`](javadoc:MethodHandles.arrayElementGetter(Class))
+method handle, to which we provide the reference to an array and the element index:
+
+```java
+MethodHandle elementGetter = MethodHandles.arrayElementGetter(String[].class);
+String element = (String) elementGetter.invoke(arr, 4); // element == "test"
+```
+
+We could also use the method handle provided by [`arrayLength`](javadoc:MethodHandles.arrayLength(Class)) to get the array
+length as integer:
+
+```java
+MethodHandle arrayLength = MethodHandles.arrayLength(String[].class);
+int length = (int) arrayLength.invoke(arr); // length == 5
+```
+
+<a id="exceptions">&nbsp;</a>
+## Exception handling
+Both `invokeExact` and `invoke` throw [`Throwable`](javadoc:Throwable), so there is no limitation to what an underlying
+method can throw. The method that invokes a method handle must either explicitly throw a `Throwable` or catch it.
+
+There are certain methods in the `MethodHandles` API that can make exception handling easier. Let's take a look at
+several examples.
+
+### `catch` wrapper
+The [`MethodHandles.catchException`](javadoc:MethodHandles.catchException(MethodHandle,Class,MethodHandle)) method can
+wrap a given method handle inside a provided exception handler method handle.
+
+Say, we have a method `problematicMethod` that performs some business logic, and a method `exceptionHandler` that handles
+a particular exception [`IllegalArgumentException`](javadoc:IllegalArgumentException). The exception handler method must
+return the same type as the original method. The first argument it accepts is a `Throwable` that we're interested in,
+after which follow the rest of the arguments that we've originally accepted:
+
+```java
+public static int problematicMethod(String argument) throws IllegalArgumentException {
+    if ("invalid".equals(argument)) {
+        throw new IllegalArgumentException();
+    }
+    return 1;
+}
+
+public static int exceptionHandler(IllegalArgumentException e, String argument) {
+    // log exception
+    return 0;
+}
+```
+
+We can look up the method handles for both these methods and wrap `problematicMethod` inside an `exceptionHandler`. The
+resulting `MethodHandle` will handle the `IllegalArgumentException` properly on invocation, continuing to throw any
+other exceptions if they arise.
+
+```java
+MethodHandles.Lookup lookup = MethodHandles.lookup();
+MethodHandle methodHandle = lookup.findStatic(Example.class, "problematicMethod", MethodType.methodType(int.class, String.class));
+MethodHandle handler = lookup.findStatic(Example.class, "exceptionHandler",
+MethodType.methodType(int.class, IllegalArgumentException.class, String.class));
```

Comment:
I think this could either be written in the same line as above (as it's the same statement) or use indentation.